openmc_plasma_source/properties.py

```diff
@@ -0,0 +1,122 @@
+"""properties
+
+A collection of utility functions for implementing class properties
+throughout openmc-plasma-source.
+"""
+
+
+def property_factory(
+    condition=None,
+    transform=None,
+    condition_err_msg: str = "",
+    transform_err_msg: str = "",
+):
+    """A generic function for creating class properties.
+
+    This function allows the user to assert a condition that must be satisfied
+    within the setter, along with the type the property should be set to. It
+    does not impose any conditions on the getter.
+
+    Parameters:
+
+        condition : Should either be None, meaning any input is accepted, or a
+            function taking one argument and returning a bool. An example of a
+            common condition may be the property must be non-negative.
```

Comment:
Maybe the function could be described as:

f(arg) --> bool instead of the current text description, what do you think?
```suggestion
        condition : Should either be None, meaning any input is accepted, or a
            function f(arg) --> bool. An example of a
            common condition may be the property must be non-negative.
```

Reply:
I agree, it is a bit clearer. I also added some extra details to those comments to hopefully clear up any other confusing aspects of them.